src/displayBackend/common/ConnectorBase.cpp

```diff
@@ -234,16 +241,29 @@ void ConnectorBase::edidPutTimings(edid* edidBlock)
 	 * We do not provide any, but detailed timings.
 	 */
 	memset(edidBlock->standard_timings, 0x01,
-		   sizeof(edidBlock->standard_timings));
+		   sizeof(edidBlock->standard_timings));	
+}
 
+void ConnectorBase::edidPutDisplayDescritor(edid* edidBlock, int descriptorIndex)
+{
 	/*
-	 * Detailed timing descriptors, in decreasing preference order,
-	 * followed by Display descriptors.
+	 * Using second descriptor as a Display descriptor to provide display name.
 	 *
-	 * We only provide a single timing here which corresponds to
-	 * XenStore configuration of this connector.
+	 * According to documentation(VESA Enhanced EDID Standard Release A, Rev.2):
+	 * Up to 13 alphanumeric characters (using ASCII codes) may be used to define the model name
+	 * of the display product. The data shall be sequenced such that the 1st byte (ASCII code) = the 1st character,
+	 * the 2nd byte (ASCII code) = the 2nd character, etc. If there are less than 13 characters in the string,
+	 * then terminate the display product name string with ASCII code ‘0Ah’ (line feed) and pad the unused bytes
+	 * in the field with ASCII code ‘20h’ (space).
 	 */
-	edidPutDetailedTiming(edidBlock, 0, mCfgWidth, mCfgHeight, EDID_DPI);
+	const char* display_name = "pv-display";
+	detailed_non_pixel* descriptor = &edidBlock->detailed_timings[descriptorIndex].data.other_data;
+	descriptor->type = EDID_DETAIL_MONITOR_NAME;
+	memset(descriptor->data.str.str, ' ', DRM_EDID_DISPLAY_NAME_MAX_LENGTH);
+	size_t length = std::min(size_t(DRM_EDID_DISPLAY_NAME_MAX_LENGTH ), strlen(display_name));
+	memcpy(descriptor->data.str.str, display_name, length);
+	if (length < DRM_EDID_DISPLAY_NAME_MAX_LENGTH)
```

Comment:
maybe it is reasonable using of strcpy

Reply:
we do not need here strcpy, since we only need a character data, but not a string delimiter.